File: Brainz/Model/ReleaseGroup.cs
namespace Brainz.Model
{
    using System;
    using Newtonsoft.Json;

    public partial class ReleaseGroup
    {
        [JsonProperty("secondary-type-ids")]
        public Guid[] SecondaryTypeIds { get; set; }

        [JsonProperty("disambiguation")]
        public string Disambiguation { get; set; }

        [JsonProperty("first-release-date")]
        public string FirstReleaseDate { get; set; }

        [JsonProperty("primary-type-id")]
        public Guid? PrimaryTypeId { get; set; }

        [JsonProperty("primary-type")]
        public string PrimaryType { get; set; }

        [JsonProperty("id")]
        public Guid Id { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("secondary-types")]
        public string[] SecondaryTypes { get; set; }
    }
}

File: go/src/model/releasegroup.go
package model

import (
	"fmt"
	"strings"
)

type ReleaseGroup struct {
	SecondaryTypeIDs []string `json:"secondary-type-ids"`
	Disambiguation   string   `json:"disambiguation"`
	FirstReleaseDate string   `json:"first-release-date"`
	PrimaryTypeID    string   `json:"primary-type-id"`
	PrimaryType      string   `json:"primary-type"`
	ID               string   `json:"id"`
	Title            string   `json:"title"`
	SecondaryTypes   []string `json:"secondary-types"`
}

func (releaseGroup ReleaseGroup) DisambiguatedTitle() string {
	disambiguation := ""

	if releaseGroup.Disambiguation != "" {
		disambiguation = fmt.Sprintf(" (%s)", releaseGroup.Disambiguation)
	}

	return fmt.Sprintf("%s%s", releaseGroup.Title, disambiguation)
}

func (releaseGroup ReleaseGroup) Types() string {
	types := []string{releaseGroup.PrimaryType}
	types = append(types, releaseGroup.SecondaryTypes[:]...)

	return strings.Join(types, ", ")
}
